File: excellence/services/api/models/table_model.py
from typing import List, Literal, Optional, Union

from .base import CamelModel


InferredType = Literal[
    "text",
    "number",
    "currency",
    "date",
    "percentage",
    "boolean",
]


class ColumnSchema(CamelModel):
    name: str
    inferred_type: InferredType
    format: Optional[str] = None


class FlaggedCell(CamelModel):
    row: int
    col: int
    raw_value: str
    suggested_value: str
    confidence: float
    reason: str


class TableModel(CamelModel):
    id: str
    name: str
    sheet_name: str
    headers: List[ColumnSchema]
    rows: List[List[Union[str, float, bool, None]]]
    flagged_cells: List[FlaggedCell]
    source_ref: str
    extracted_at: int

File: excellence/packages/extraction-core/src/models/TableModel.ts
export type InferredType =
	| 'text'
	| 'number'
	| 'currency'
	| 'date'
	| 'percentage'
	| 'boolean';

export interface ColumnSchema {
	name: string;
	inferredType: InferredType;
	format?: string;
}

export interface FlaggedCell {
	row: number;
	col: number;
	rawValue: string;
	suggestedValue: string;
	confidence: number;
	reason: string;
}

export interface CellData {
	text: string;
	confidence: number;
	bbox: [number, number, number, number];
}

export interface TableModel {
	id: string;
	name: string;
	sheetName: string;
	headers: ColumnSchema[];
	rows: Array<Array<string | number | boolean | null>>;
	flaggedCells: FlaggedCell[];
	sourceRef: string;
	extractedAt: number;
}
